Orbit.h

```diff
@@ -1,122 +1,104 @@
+#define ORBIT_SCOPE(name) orbit::Scope ORBIT_UNIQUE(ORB)(ORBIT_LITERAL(name))
+
+// ORBIT_START/ORBIT_STOP: profile time span on a single thread.
+#define ORBIT_START(name) orbit::Start(ORBIT_LITERAL(name))
+#define ORBIT_STOP orbit::Stop()
+
+// ORBIT_START_ASYNC/ORBIT_STOP_ASYNC: profile time span across threads.
+#define ORBIT_START_ASYNC(name, id) orbit::StartAsync(ORBIT_LITERAL(name), id)
+#define ORBIT_STOP_ASYNC(id) orbit::StopAsync(id)
+
+// ORBIT_[type]: graph variables.
+#define ORBIT_INT(name, value) orbit::TrackInt(ORBIT_LITERAL(name), value)
+#define ORBIT_INT64(name, value) orbit::TrackInt64(ORBIT_LITERAL(name), value)
+#define ORBIT_UINT(name, value) orbit::TrackUint(ORBIT_LITERAL(name), value)
+#define ORBIT_UINT64(name, value) orbit::TrackUint64(ORBIT_LITERAL(name), value)
+#define ORBIT_FLOAT(name, value) orbit::TrackFloat(ORBIT_LITERAL(name), value)
+#define ORBIT_DOUBLE(name, value) orbit::TrackDouble(ORBIT_LITERAL(name), value)
 
-//-----------------------------------------------------------------------------
+#else
+
+#define ORBIT_SCOPE(name)
+#define ORBIT_START(name)
+#define ORBIT_STOP
+#define ORBIT_START_ASYNC(name, id)
+#define ORBIT_STOP_ASYNC(id)
+#define ORBIT_INT(name, value)
+#define ORBIT_INT64(name, value)
+#define ORBIT_UINT(name, value)
+#define ORBIT_UINT64(name, value)
+#define ORBIT_FLOAT(name, value)
+#define ORBIT_DOUBLE(name, value)
+
+#endif
+
+// NOTE: Do not use any of the code below directly.
+
+// Internal macros.
+#define ORBIT_LITERAL(x) ("" x)
 #define ORBIT_CONCAT_IND(x, y) (x##y)
 #define ORBIT_CONCAT(x, y) ORBIT_CONCAT_IND(x, y)
 #define ORBIT_UNIQUE(x) ORBIT_CONCAT(x, __COUNTER__)
+#define ORBIT_UNUSED(x) (void)(x)
+#define ORBIT_NOOP       \
+  static volatile int x; \
+  x;
 
-//-----------------------------------------------------------------------------
-void OrbitAPI::Init() {
-  HINSTANCE& dllHandle = GetHandle();
-  if (!dllHandle) {
-    const TCHAR* dllName =
-        sizeof(size_t) == 4 ? TEXT("Orbit32.dll") : TEXT("Orbit64.dll");
-    if (dllHandle = LoadLibrary(dllName)) {
-      GetInitRemote() =
-          (InitRemote)GetProcAddress(dllHandle, "OrbitInitRemote");
-    }
-  }
-}
+#if _WIN32
+#define NO_INLINE __declspec(noinline)
+#else
+#define NO_INLINE __attribute__((noinline))
+#endif
 
-//-----------------------------------------------------------------------------
-bool OrbitAPI::Connect(const char* a_Host, int a_Port) {
-  Init();
-  if (GetHandle() && GetInitRemote()) {
-    char host[256] = {0};
-    sprintf_s(host, sizeof(host), "%s:%i", a_Host, a_Port);
-    GetInitRemote()(host);
-    return IsConnected();
-  }
-
-  return false;
-}
+namespace orbit {
 
-//-----------------------------------------------------------------------------
-bool OrbitAPI::IsConnected() {
-  return GetIsConnected() ? GetIsConnected()() : false;
-}
+// NOTE: Do not use these directly, use corresponding macros instead.
+inline void NO_INLINE Start(const char*) { ORBIT_NOOP; }
```

Comment:
I might be missing something but why both `inline` and `NO_INLINE`?

Reply:
NO_INLINE because those functions need to be callable so that we can instrument them.  "Inline" so that the linker doesn't complain about multiple definitions for the function.  Note that the ORBIT_NOOP is there so that the compiler doesn't strip the empty function altogether.

Reply:
Ah right, thanks.